### PR TITLE
Support EL in resource name

### PR DESCRIPTION
**Pull request by brightlizard**
https://github.com/gravitee-io/issues/issues/2629

### CODE REVIEW

File: src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java [hunk 1]
```diff
@@ -81,6 +81,15 @@ public Oauth2Policy (OAuth2PolicyConfiguration oAuth2PolicyConfiguration) {
     public void onRequest(Request request, Response response, ExecutionContext executionContext, PolicyChain policyChain) {
         logger.debug("Read access_token from request {}", request.id());
 
+        final String resourceNameTemplate = oAuth2PolicyConfiguration.getOauthResource();
+
+        if(resourceNameTemplate != null){
```
Comment (brasseld) [inline src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java]: I do not see any reason for which the `resourceNameTemplate` would be null. Can you explain a bit please?

**Comment by brasseld** [comment]
Thanks for the PR @brightlizard 

Can you please amend your commit and add a reference to the issue in commit message?

File: src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java (hunk 1 above)
Comment (brightlizard) [reply src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java]: It's only for passing tests. It can be null in some tests.

Comment (brasseld) [reply src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java]: As per https://github.com/gravitee-io/gravitee-policy-oauth2/blob/master/src/main/resources/schemas/schema-form.json#L45 resource is required so I assume tests are wrong and resource is always provided.

Comment (brightlizard) [reply src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java]: Yes, right, I see now that I can fix tests and reduce code in policy. I will update PR a little bit later.

**Comment by brasseld** [comment]
@brightlizard please squash the three commits to a single one.

**Comment by brightlizard** [comment]
> Please squash the four commits to a single one.

Hi again. I joined all commits. But unfourtunately I reopened the PR unintentional.

**Comment by brightlizard** [comment]
Please review this PR

**Pull request merged**